cron/internal/data/iterator.go

```diff
@@ -47,10 +52,29 @@ type csvIterator struct {
 	decoder *csvutil.Decoder
 	err     error
 	next    RepoFormat
+	count   int
+}
+
+// check if the most recently decoded record is a header, and skip it if it is.
+func (reader *csvIterator) ignoreHeader() {
+	header, err := csvutil.Header(RepoFormat{}, "csv")
+	if err != nil {
+		reader.err = err
+		return
+	}
+	lastRead := reader.decoder.Record()
+	if reflect.DeepEqual(header, lastRead) {
+		reader.err = reader.decoder.Decode(&reader.next)
+	}
 }
 
 func (reader *csvIterator) HasNext() bool {
 	reader.err = reader.decoder.Decode(&reader.next)
+	// only compares the first read against the csv header for performance
+	if reader.count == 0 {
+		reader.ignoreHeader()
+	}
+	reader.count++
```

Comment:
```suggestion
	reader.err = reader.ignoreIfHeader(); 
	reader.err = reader.decoder.Decode(&reader.next)
```

Reply:
I'll make it similar to this, but I'll need to swap the order of the two calls back to what I had since `reader.ignoreIfHeader()` relies on `decoder.Decode` being called.

Reply:
My suggestion was to avoid the `decoder.Decode` call inside `ignoreIfHeader`. Could `ignoreIfHeader` simply `return reflect.DeepEqual()` ?

Reply:
Or maybe, this should be:

```
if header, reader.err := reader.isHeader(); header {
  _ = reader.decoder.Decode(nil) // ignore header
}
reader.err = reader.decoder.Decode(&reader.next)
```

Wdyt?

Reply:
Sorry, I see what you mean. Ignore me.

Reply:
```go
// returns true on the first call if the most recently decoded record is a header.
// always returns false on subsequent calls for performance.
func (reader *csvIterator) isHeader() bool {
	if reader.afterHeader {
		return false
	}
	header, err := csvutil.Header(RepoFormat{}, "csv")
	if err != nil {
		reader.err = err
		return false
	}
	lastRead := reader.decoder.Record()
	reader.afterHeader = true
	return reflect.DeepEqual(header, lastRead)
}

func (reader *csvIterator) HasNext() bool {
	reader.err = reader.decoder.Decode(&reader.next)
	if reader.isHeader() {
		reader.err = reader.decoder.Decode(&reader.next)
	}
	return !errors.Is(reader.err, io.EOF)
}
```
something like that?

Reply:
Yep, that LGTM